src/Languages/Core/Impl/Formatting/IndentBuilder.cs

```diff
@@ -233,5 +233,17 @@ public IndentBuilder(IndentType indentType, int indentSize, int tabSize)
 
             return indent;
         }
+
+        /// <summary>
+        /// Determines indentation based on the leading whitespace in the current line.
+        /// </summary>
+        public static int GetLineIndentSize(TextBuilder tb, int position, int tabSize) {
```

Comment:
This should probably check `position` for validity, and assert or throw if it's 0.